Modules/mathmodule.c

```diff
@@ -1063,8 +1064,7 @@ math_1_to_whatever(PyObject *arg, double (*func) (double),
     errno = 0;
     r = (*func)(x);
     if (Py_IS_NAN(r) && !Py_IS_NAN(x)) {
-        PyErr_SetString(PyExc_ValueError,
-                        "math domain error"); /* invalid arg */
+        PyErr_Format(PyExc_ValueError, err_msg, arg); /* invalid arg */
```

Comment:
Shouldn't same message be used below on the line 1076 as well (infinite r for finite input)?  atanh(-1) is an example.

Reply:
> Shouldn't same message be used below on the line 1076 as well (infinite r for finite input)? atanh(-1) is an example.

I was wondering if there is a function that can have both input and output( NAN r for non-NAN x and infinite r for finite x). 

If such a function exists, it may lead to confusing error messages, because one error message corresponds to two scenarios. I haven't thought of such a function yet, please suggest if it exists. :)

Reply:
Same atanh :)

atanh(2) - for the 1st scenario and atanh(1) - for the second.  I doubt it will be confusing if you raise instead something like ValueError("atanh expects an argument from (-1, -1), got %R").  I don't think it does make sense to have a different descriptions for pole errors (like atanh(1)).  @mdickinson ?

Reply:
See [9c32ae0](https://github.com/python/cpython/pull/102523/commits/9c32ae022f07ea90892175a2d7f5c777621be0e0) from #102523.